core/src/software/aws/toolkits/core/telemetry/MetricEvent.kt

```diff
@@ -10,13 +10,15 @@ import java.time.Instant
 import software.amazon.awssdk.services.toolkittelemetry.model.Unit as MetricUnit
 
 interface MetricEvent {
-    val namespace: String
+    @Deprecated("namespace should not be used")
```

Comment:
Alternatively, should we deprecate this whole file and have the TelemetryService record individual `Datum` that are not encapsulated by a `MetricEvent`?

Reply:
What does it mean to have more than one datum with the same createTime?

Reply:
Is there a drawback of recording individual Datum?

Reply:
Data with the same createTime results in each datum with the same timestamp when recorded in the back-end. I see the whole class as a convenience, so there is no downside in recording datum individually.

Reply:
I'm not seeing the value in sending Datum with a matching timestamp. I'm neutral on the initial question.

Reply:
If we're planning on eventually codegen-ing our metric classes, we're going to throw this whole file away anyways, so I'm inclined to merge as-is.